Homework-3 (course-2)

DuDaria opened the pull request.

DuDaria committed:
Homework-3 (course-2)

DuDaria committed:
Homework-3(new) (course-2)

Исправила ошибки. Добавила transition на страницы

tahavn commented on src/components/breadcrumbs/_breadcrumbs.scss:
```diff
@@ -8,23 +8,24 @@
 
-.breadcrumbs__current-page {
-  opacity: 0.5;
+  &__current-page {
```
здесь лучше использовать класс как у других элементов breadcrumbs__link
а  для активной страницы добавить модификатор breadcrumbs__link_current

+ странная структура breadcrumbs, вставлять разделитель как img странно и неудобно, лучше для разделителя использовать псевдоэлемент

tahavn commented on src/pages/toys/index.html:
```diff
@@ -22,96 +22,97 @@ <h1 class="title container">TOY FOR KIDS</h1>
+          <fieldset class="color"> 
+            <span class="color__title">Color</span>
+            <ul class="color__list">
+              <li class="color__list-etem">
```
вполне можно использовать color__item в бэм иногда приходиться делать длинные имена классов, но лучше этого избегать если есть такая возможность.

tahavn commented on src/pages/toys/index.html:
```diff
@@ -22,96 +22,97 @@ <h1 class="title container">TOY FOR KIDS</h1>
+          <fieldset class="delivery">
+            <span class="delivery__title">Delivery</span>
+            <ul class="delivery__list">
+              <li class="delivery__list-etem">
```
вполне можно использовать delivery__item в бэм иногда приходиться делать длинные имена классов, но лучше этого избегать если есть такая возможность.

tahavn commented on src/pages/toys/index.html:
```diff
@@ -22,96 +22,97 @@ <h1 class="title container">TOY FOR KIDS</h1>
+              <li class="color__list-etem">
+                <label class="color__label">
+                  <input class="color__input hidden" type="checkbox" name="color-black" value="black">
+                  <span class="color__click color__click_active"></span>
```
в данном случае модификатор лишний, при выборе другого элемента класс модификатора останется на этом элементе без изменений.  при использовании radio и chekbox неактуально использовать модификаторы, если вы не используете переключение класса через js

tahavn commented on src/pages/toys/index.html:
```diff
@@ -22,96 +22,97 @@ <h1 class="title container">TOY FOR KIDS</h1>
+              <li class="delivery__list-etem">
+                <label class="delivery__label">
+                  <input class="delivery__input hidden" type="radio" name="shipping" value="fast">
+                  <span class="delivery__click delivery__click_active"></span>
```
в данном случае модификатор лишний, при выборе другого элемента класс модификатора останется на этом элементе без изменений.  при использовании radio и chekbox неактуально использовать модификаторы, если вы не используете переключение класса через js

tahavn commented on src/pages/toys/index.html:
```diff
@@ -22,96 +22,97 @@ <h1 class="title container">TOY FOR KIDS</h1>
+              <li>
+                <label class="amount__label">
+                  <input class="amount__input hidden" type="radio" name="range" value="0-5">
+                  <span class ="amount__text amount__text_active">0-5</span>
```
в данном случае модификатор лишний, при выборе другого элемента класс модификатора останется на этом элементе без изменений.  при использовании radio и chekbox неактуально использовать модификаторы, если вы не используете переключение класса через js

tahavn commented on src/pages/toys/index.html:
```diff
@@ -124,78 +125,57 @@ <h2 class="hidden">Goods</h2>
           <li class="goods__list-item goods__list-item_1">
```
1. goods__list-item_1 _2  _3 и т.д. с какой целью у этих элементов модификаторы? некорректное имея модификатора, имя должно должно быть интуитивно понятно, чтобы сразу было ясно для чего добавлен модификатор. касается всех

2. вполне можно использовать goods__item в бэм иногда приходиться делать длинные имена классов, но лучше этого избегать если есть такая возможность. это касается всех страниц где встречается этот элемент

tahavn commented on src/pages/toys/index.html:
```diff
@@ -124,78 +125,57 @@ <h2 class="hidden">Goods</h2>
               <img src="/../../static/images/sharkey.jpg" alt="SHARKY BEAN BAG CHAIR COVER" class="goods__link-bg">
-              <span class="goods__hover"></span>
+              <span class="goods__hover">LEAN MORE</span>
               <span class="goods__link-title">SHARKY BEAN BAG CHAIR COVER</span>
```
вполне можно использовать goods__link в бэм иногда приходиться делать длинные имена классов, но лучше этого избегать если есть такая возможность. то касается всех страниц где встречается этот элемент

tahavn commented on src/pages/main/index.html:
```diff
@@ -65,17 +66,17 @@ <h2 class="goods__title">CATEGORY PRODUCTS</h2>
-        <div class="goods__slider">
-          <div class="goods__slider-nav">
-            <button type="button" class="goods__slider-prev">
+        <div class="goods-slider__slider">
```
у вас нет блока goods-slider. goods-slider__slider и другие элементы ниже подобной серии нарушают бэм. элемент не может быть без блока.

здесь было корректно дать класс goods-slider если хотите создать отдельный блок. или goods__slider если хотите создать элемент для блока goods

tahavn commented on src/pages/main/index.html:
```diff
@@ -65,17 +66,17 @@ <h2 class="goods__title">CATEGORY PRODUCTS</h2>
+        <div class="goods-slider__slider">
+          <div class="goods-slider__slider-nav">
+            <button class="goods-slider__button-prev" type="button" >
               <img class="img-button" src="/../../static/images/vector-prev.png" alt="button-prev">
             </button>
-            <button type="button" class="goods__slider-next">
+            <button class="goods-slider__button-next" type="button">
               <img class="img-button" src="/../../static/images/vector-next.png" alt="button-next">
             </button>
           </div>
 
-          <div class="goods__slider-content">
+          <div class="goods-slider__slider-content">
```
не корректные классы, см. примечание выше

tahavn commented on src/pages/main/index.html:
```diff
@@ -111,30 +113,30 @@ <h2 class="additional-inform__title">DELIVERY</h2>
           <li>
-            <a class="partner__list-item" href="#">
-              <img class="partner__img" src="/../../static/images/mail.png" alt="mail">
+            <a class="partners__list-item" href="#">
```
вполне можно использовать partners__item в бэм иногда приходиться делать длинные имена классов, но лучше этого избегать если есть такая возможность.

tahavn commented on src/pages/main/index.html:
```diff
@@ -149,7 +151,7 @@ <h3 class="about-us__title">DESIGN LIFE KIDS</h3>
         </div>
 
-        <div class="about-us__block inform-2">
+        <div class="about-us__block-inform-2">
           <h3 class="about-us__title">CONTACTS</h3>
           <p class="about-us__text">
             Some days i feel like i've been working on dlk my entire life and 
```
вполне можно было это сделать отдельным блоком contacts 
и не давать странный класс about-us__block inform-1  -2

tahavn commented on src/pages/main/index.html:
```diff
@@ -65,17 +66,17 @@ <h2 class="goods__title">CATEGORY PRODUCTS</h2>
-            <button type="button" class="goods__slider-prev">
+        <div class="goods-slider__slider">
+          <div class="goods-slider__slider-nav">
+            <button class="goods-slider__button-prev" type="button" >
```
лучше использовать класс goods-slider__button с модификатором goods-slider__button_prev и goods-slider__button_next соответственно

tahavn commented on src/pages/blog/index.html:
```diff
@@ -25,7 +25,7 @@ <h2 class="hidden">announce</h2>
         <ul class="announce__list">
           <li class="announce__list-item">
```
вполне можно использовать announce__item в бэм иногда приходиться делать длинные имена классов, но лучше этого избегать если есть такая возможность.

tahavn commented on src/components/header/index.html:
```diff
@@ -1,18 +1,18 @@
   <div class="header container">
-    <a class="header__logo no_active" href="/">
+    <a class="header__logo no-active" href="/">
       <img class="header__img-logo" src="/static/images/logo.png" alt="logo">
```
для соблюдение последовательности класс header__logo-img подходит больше

tahavn commented on src/components/header/index.html:
```diff
@@ -1,18 +1,18 @@
-        <li><a class="header__nav-item" href="/pages/blog/index.html">BLOG</a></li>
-        <li><a class="header__nav-item" href="/pages/catalog/index.html">CATALOG</a></li>
-        <li><a class="header__nav-item" href="/pages/toys/index.html">TOYS</a></li>
+        <li><a class="header__nav-item no-active-1" href="/pages/blog/index.html">BLOG</a></li>
+        <li><a class="header__nav-item no-active-2" href="/pages/catalog/index.html">CATALOG</a></li>
+        <li><a class="header__nav-item no-active-3" href="/pages/toys/index.html">TOYS</a></li>
```
странные класс no-active-1-2-3  особенно если учесть, что ни никак не используются для стилизации

tahavn reviewed (changes requested):
Правки ДЗ-3

DuDaria committed:
Update Homework 3 (course-2), Homework 4 (course-2), Homework 5 (course-2)

Исправила замечания по Дз-3. Сделала Дз-4 и Дз-5. SVG-анимация была по желанию, пока еще не создала.

DuDaria committed:
Update Homework 4 (course-2)

Исправила движение машин, добавила облака

DuDaria committed:
SVG-Animation (Homework 5)

Добавила доп страницу, сделала анимацию SVG

tahavn reviewed (commented):
дз-3

tahavn reviewed (commented):
дз
